parlai/core/params.py

```diff
@@ -127,22 +150,24 @@ def add_model_args(self, args=None):
             if hasattr(agent, 'add_cmdline_args'):
                 agent.add_cmdline_args(self)
             if hasattr(agent, 'dictionary_class'):
-                self.add_argument(
-                    '--dict-class', default=agent.dictionary_class(), type=str,
-                    help='the class of the dictionary agent used')
+                model_args.set_defaults(dict_class=agent.dictionary_class())
```

Comment:
you actually put a whole class in the opt?  this seems really really weird. . can't it just be the name?